subprojects/core-api/src/main/java/org/gradle/api/tasks/TaskContainer.java

```diff
@@ -261,7 +261,7 @@
      * @throws NullPointerException If any of the values in {@code constructorArgs} is null.
      * @since 4.9
      */
-    <T extends Task> TaskProvider<T> register(String name, Class<T> type, Object... constructorArgs) throws InvalidUserDataException;
+    <T extends Task> TaskProvider<T> register(String name, Class<T> type, @Nullable Object... constructorArgs) throws InvalidUserDataException;
```

Comment:
IIUC, this is saying that arguments in `constructorArgs` can be null?

Reply:
Correct, yes. 😊

Reply:
`@Nullable Object... constructorArgs` means non-nullable array with `@Nullable Object elements`
`Object @Nullable... constructorArgs` means `@Nullable array` with to non-null Object elements
`@Nullable Object @Nullable... constructorArgs` means `@Nullable array` with `@Nullable Object elements`

Reply:
Gotta hate Java, fixing.

Reply:
OK, then this isn't going to work, sorry. We don't allow null elements in constructorArgs for reasons involving how we identify which constructor to use.

Reply:
This should be at least documented then, because it's not obvious. 😉